File: 3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.java
class Solution {
    public String shortestBeautifulSubstring(String s, int k) {

        String res = "";
        int one = 0;
        int l = 0;

        for (int r = 0; r < s.length(); r++) {

            if (s.charAt(r) == '1') {
                one++;
            }

            while (one > k) {
                if (s.charAt(l) == '1') {
                    one--;
                }
                l++;
            }

            if (one == k) {

                while (s.charAt(l) == '0') {
                    l++;
                }

                String curr = s.substring(l, r + 1);

                if (res.isEmpty() ||
                    curr.length() < res.length() ||
                    (curr.length() == res.length() &&
                     curr.compareTo(res) < 0)) {

                    res = curr;
                }
            }
        }

        return res;
    }
}

File: 3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        res = []

        for l in range(len(s)):
            one = 0

            for r in range(l,len(s)):

                if s[r] == "1":
                    one += 1
                
                if one == k:
                    res.append(s[l:r+1])
                    break

                if one > k :
                    break

        if not res:
            return ""

        min_len = min(map(len, res))

        return min(x for x in res if len(x) == min_len)

                
        